Too many Compilation Warnings generated in esolver module

### Describe the Code Quality Issue

Many warnings are generated during compilation, disrupting the normal process of locating compilation errors in development.
```cpp
In file included from /abacus-develop/source/module_io/input_conv.cpp:8:
In file included from /abacus-develop/source/module_hamilt_general/module_surchem/surchem.h:11:
In file included from /abacus-develop/source/module_hamilt_pw/hamilt_pwdft/global.h:9:
In file included from /abacus-develop/source/module_relax/relax_driver.h:5:
/abacus-develop/source/module_esolver/esolver_ks.h:73:22: warning: 'after_scf' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
   73 |         virtual void after_scf(const int istep);
      |                      ^
/abacus-develop/source/module_esolver/esolver_ks_lcao.h:25:32: note: in instantiation of template class 'ModuleESolver::ESolver_KS<std::complex<double>>' requested here
   25 | class ESolver_KS_LCAO : public ESolver_KS<TK> {
      |                                ^
/abacus-develop/source/module_esolver/esolver_ks_lcao_tddft.h:12:38: note: in instantiation of template class 'ModuleESolver::ESolver_KS_LCAO<std::complex<double>, double>' requested here
   12 | class ESolver_KS_LCAO_TDDFT : public ESolver_KS_LCAO<std::complex<double>, double>
      |                                      ^
/abacus-develop/source/module_esolver/esolver_fp.h:68:22: note: overridden virtual function is here
   68 |         virtual void after_scf(const int istep);
      |                      ^
```
They are related to 'after_scf' override in esolver.
I kindly request that the development team reviews the compilation warnings and takes necessary action to resolve or reduce them.

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific code file or section with the code quality issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Refactor the code to improve code quality, following the suggested solution.
- [ ] Ensure the refactored code adheres to the project's coding standards.
- [ ] Test the refactored code to ensure it functions as expected.
- [ ] Update any relevant documentation, if necessary.
- [ ] Submit a pull request with the refactored code and a description of the changes made.

--- a/source/module_esolver/esolver_ks.h
+++ b/source/module_esolver/esolver_ks.h
@@ -70,7 +70,7 @@ class ESolver_KS : public ESolver_FP
         virtual void iter_finish(int& iter);
 
         //! Something to do after SCF iterations when SCF is converged or comes to the max iter step.
-        virtual void after_scf(const int istep);
+        virtual void after_scf(const int istep) override;
 
         //! <Temporary> It should be replaced by a function in Hamilt Class
 		virtual void update_pot(const int istep, const int iter) {};

